flow/connectors/utils/monitoring/monitoring.go

```diff
@@ -306,13 +306,18 @@ func (c *CatalogMirrorMonitor) UpdateStartTimeForPartition(
 	ctx context.Context,
 	runUUID string,
 	partition *protos.QRepPartition,
+	overrideStartTime *time.Time,
 ) error {
 	if c == nil || c.catalogConn == nil {
 		return nil
 	}
+	startTime := time.Now()
+	if overrideStartTime != nil {
+		startTime = *overrideStartTime
+	}
```

Comment:
```suggestion
        var startTime Time
	if overrideStartTime == nil {
		startTime := time.Now()
	} else {
		startTime = *overrideStartTime
	}
```
avoids calling `time.Now()` unnecessarily